documentation/decisions/adr/2023-12-15-deliverable-reporting-strategy.md

```diff
@@ -0,0 +1,371 @@
+
+### Limiting the 30k deliverables included in reporting
+
+Our recommended approach to limiting the deliverables that are included in reporting should consider the following criteria:
+
+- The logic for including a 30k deliverable in reporting should be clear and easy to understand
+- Users should be able to add or remove a given 30k deliverable from reports without having to change the underlying logic
+- If necessary, users should be able to change the reporting logic without rewriting significant sections of the source code
+
+## Options considered
+
+### Assigning issues to deliverables
+
+- **Milestones per 10k:** Add the issue to a milestone that represents a 10k deliverable and tag the 30k deliverable in the body of the milestone
+- **Milestones per 30k:** Add the issue to a milestone that represents a 30k deliverable
+- **Labels:** Tag the issue with a label that represents a 30k deliverable
+- **Deliverable column:** Add a custom "deliverable" column to the GitHub projects with values representing each 30k deliverable
+- **Reserved phrase:** Use a reserved phrase to tag a 30k deliverable in the body of the issue
+
+### Limiting the 30k deliverables included in reporting
+
+- **Label:** Use a label to indicate when a deliverable should be included in reporting
+- **Status:** Use the status of the deliverable in the provisional roadmap project
+- **GitHub action:** Explicitly list the deliverables to include in the GitHub action for reporting
+
+## Decision outcomes <!-- REQUIRED -->
+
+### Assigning issues to deliverables
+
+We've decided to use a "deliverable" column to assign issues to a parent 30k deliverable. This involves:
+
+- creating a single select column in both the product roadmap and the sprint board
+- populating that column with options that represent each 30k deliverables in our roadmap
+- using that column to indicate the 30k deliverable that an issue is assigned to
+
+- **Positive outcomes**
+  - We can filter, group, and sort issues by deliverable in GitHub projects and GitHub's automated insight reporting.
+  - The engineering team can continue to use GitHub milestones to organize issues into units of work that are smaller than a 30k deliverable. **Note:** While enabling the engineering team to use milestones to organize issues into smaller units of work, we want to avoid situations in which a given milestone contains issues from multiple 30k deliverables -- we can enforce this with a linter as well as a view in the GitHub project.
+  - The logic for grouping issues by deliverable will be consistent between GitHub insight reports and our custom-built reports.
+- **Negative outcomes**
+  - We won't be able to filter for issues assigned to a given 30k deliverable within the GitHub repository.
+  - Updating a field on a project can be done from the issue page itself, but the issue first needs to be added to the project before the field is updated. This requires an extra step compared to assigning a label or milestone.
```

Comment:
[non-blocking comment] Is it possible to add the field in to the issue template when creating an issue?

Reply:
We could add a field to the template, but the value that folks enter would only show up in the body of the issue, not the project fields.

If we track the associated deliverable in the body of the issue and in the project column, I'd be a bit worried about discrepancies if we were to reassign the issue to another deliverable, but forgot to change the body of the issue. Part of the motivation for this ADR is to limit the "sources of truth" for associating issues with deliverables.

Reply:
ah, yeah, agreed that we wouldn't want it only in the body of the issue. thanks, billy!